htlcswitch/link_test.go

```diff
@@ -277,125 +275,93 @@ func TestChannelLinkBidirectionalOneHopPayments(t *testing.T) {
 	}
 	defer cleanUp()
 
-	n := newThreeHopNetwork(t, channels.aliceToBob, channels.bobToAlice,
-		channels.bobToCarol, channels.carolToBob, testStartingHeight)
+	n := newTwoHopNetwork(
+		t, channels.aliceToBob, channels.bobToAlice, testStartingHeight,
+	)
 	if err := n.start(); err != nil {
 		t.Fatal(err)
 	}
 	defer n.stop()
-	bobBandwidthBefore := n.firstBobChannelLink.Bandwidth()
-	aliceBandwidthBefore := n.aliceChannelLink.Bandwidth()
-
-	debug := false
-	if debug {
-		// Log message that alice receives.
-		n.aliceServer.intersect(createLogFunc("alice",
-			n.aliceChannelLink.ChanID()))
-
-		// Log message that bob receives.
-		n.bobServer.intersect(createLogFunc("bob",
-			n.firstBobChannelLink.ChanID()))
-	}
 
+	// Fill up the commitment from Alice's side with 20 sat payments.
+	count := (input.MaxHTLCNumber / 2)
 	amt := lnwire.NewMSatFromSatoshis(20000)
 
 	htlcAmt, totalTimelock, hopsForwards := generateHops(amt,
-		testStartingHeight, n.firstBobChannelLink)
-	_, _, hopsBackwards := generateHops(amt,
-		testStartingHeight, n.aliceChannelLink)
+		testStartingHeight, n.bobChannelLink)
 
-	type result struct {
-		err    error
-		start  time.Time
-		number int
-		sender string
-	}
+	firstHop := n.aliceChannelLink.ShortChanID()
+
+	// Create channels to buffer the preimage and error channels used in
+	// making the preliminary payments.
+	alicePreimageChan := make(chan lntypes.Preimage, count)
+	aliceErrChan := make(chan chan error, count)
 
-	// Send max available payment number in both sides, thereby testing
-	// the property of channel link to cope with overflowing.
-	count := 2 * input.MaxHTLCNumber
-	resultChan := make(chan *result, count)
-	for i := 0; i < count/2; i++ {
+	var wg sync.WaitGroup
+	for i := 0; i < count; i++ {
+		var preimage lntypes.Preimage
+		_, err := rand.Read(preimage[:])
```

Comment:
Could use deterministic preimages like `lntypes.Preimage{byte(i>>8), byte(i)}`. Not very important, more a guideline thing

Reply:
done